Fix WHO not respecting feature HIS_USERIP and showing IP to ordinary users.

diff -r 059a1d81b9fb ircd/whocmds.c
--- a/ircd/whocmds.c	Wed Jul 03 19:00:21 2013 +0200
+++ b/ircd/whocmds.c	Thu Jul 04 16:42:59 2013 +0200
@@ -129,7 +129,8 @@
 
   if (fields & WHO_FIELD_NIP)
   {
-    const char* p2 = (HasHiddenHost(acptr) || HasSetHost(acptr)) && !IsAnOper(sptr) ?
+    /* hide IP when feature HIS_USERIP is enabled - snircd */
+    const char* p2 = (HasHiddenHost(acptr) || HasSetHost(acptr) || feature_bool(FEAT_HIS_USERIP)) && !IsAnOper(sptr) ?
       feature_str(FEAT_HIDDEN_IP) :
       ircd_ntoa(&cli_ip(acptr));
     *(p1++) = ' ';
